dart/biomechanics/DynamicsFitter.cpp

```diff
@@ -13134,108 +13144,196 @@ void DynamicsFitter::recomputeGRFs(
+        Eigen::Index closestFoot;
+        sumSquaredDistances.minCoeff(&closestFoot);
+
+        // std::cout << "Force Plate " << i << " time range [" <<
+        // lastStartedTrack
+        //           << "," << (poses.cols() - 1)
+        //           << "] Closest foot: " << closestFoot << " with distances "
+        //           << sumSquaredDistances.transpose() << std::endl;
+
+        // Now assign everything in that track to the foot that was closest
+        // over the whole track
+        for (int assignT = lastStartedTrack; assignT < poses.cols(); assignT++)
         {
-          Eigen::Index closestFoot;
-          sumSquaredDistances.minCoeff(&closestFoot);
-
-          // std::cout << "Force Plate " << i << " time range ["
-          //           << lastStartedTrack << "," << (t - 1)
-          //           << "] Closest foot: " << closestFoot << " with
-          //           distances
-          //           "
-          //           << sumSquaredDistances.transpose() << std::endl;
-
-          // Now assign everything in that track to the foot that was closest
-          // over the whole track
-          for (int assignT = lastStartedTrack; assignT < t; assignT++)
+          // This is a map of [trial][forcePlate][timestep]
+          init->forcePlatesAssignedToContactBody[trial][i][assignT]
+              = closestFoot;
+        }
+      }
+
+      // 2.2. Check if this force plate is only ever assigned to a single foot
+      // (which is great) and if so, then assign all timesteps to that foot.
+      std::vector<int> assignedToFeet;
+      for (int t = 0; t < poses.cols(); t++)
+      {
+        int closestFoot = init->forcePlatesAssignedToContactBody[trial][i][t];
+        if (closestFoot != -1)
+        {
+          // If closestFoot is already in the list, add it to the list
+          if (std::find(
+                  assignedToFeet.begin(), assignedToFeet.end(), closestFoot)
+              == assignedToFeet.end())
           {
-            // This is a map of [trial][forcePlate][timestep]
-            init->forcePlatesAssignedToContactBody[trial][i][assignT]
-                = closestFoot;
-
-            Eigen::Vector3s force = forcePlates[i].forces[assignT];
-            Eigen::Vector3s cop = forcePlates[i].centersOfPressure[assignT];
-            Eigen::Vector3s moments = forcePlates[i].moments[assignT];
-
-            Eigen::Vector6s wrench = Eigen::Vector6s::Zero();
-            wrench.head<3>() = moments;
-            wrench.tail<3>() = force;
-            Eigen::Isometry3s wrenchT = Eigen::Isometry3s::Identity();
-            wrenchT.translation() = cop;
-            Eigen::Vector6s worldWrench = math::dAdInvT(wrenchT, wrench);
-
-            // If multiple force plates assign to the same foot, sum up the
-            // forces
-            init->grfTrials[trial].block<6, 1>(closestFoot * 6, assignT)
-                += worldWrench;
+            assignedToFeet.push_back(closestFoot);
```

Comment:
Can this `push_back` ever be reached? It looks like `assignedToFeet` gets initialized above, so it should always be empty the first time it hits this if-statement, and the if-statement is only true if `closestFoot` is already in the vector, but I don't see how that's possible before this line. Maybe I'm missing something.

Reply:
Oh, sorry, I made a dumb typo. If it's _not_ already in the list, it gets added to the list.